src/src/PluginStructs/P164_data_struct.cpp

```diff
@@ -0,0 +1,889 @@
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Plugin data structure for P164 "GASES - ENS16x (TVOC, eCO2)"
+// Plugin for ENS160 & ENS161 TVOC and eCO2 sensor with I2C interface from ScioSense
+// Based upon: https://github.com/sciosense/ENS160_driver
+// For documentation see 
+// https://www.sciosense.com/wp-content/uploads/documents/SC-001224-DS-9-ENS160-Datasheet.pdf
+//
+// 2023 By flashmark
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "../PluginStructs/P164_data_struct.h"
+
+#ifdef USES_P164
+
+// A curious delay inserted in the original code [ms] 
+#define ENS160_BOOTING          10
+
+// ENS160 registers for version V0
+#define ENS160_REG_PART_ID      0x00    // 2 byte register for part identification
+#define ENS160_REG_OPMODE       0x10    // Operation mode register
+#define ENS160_REG_CONFIG       0x11    // Pin configuration register
+#define ENS160_REG_COMMAND      0x12    // Additional system commands
```

Comment:
#define stuff like this is, by convention, placed in the accompanying .h file...

Reply:
Well.. unless it is only used in the .cpp.
Then it might be useful to only define it in the .cpp.

Things to keep in mind are:
- Can it conflict somewhere else the .h file is included?
- Are related defines needed elsewhere, then keep them together
- Does the .h file get included a lot? If so, then having more in the .h file might add to the compile duration

Reply:
I started with Sciosense original software. Decided to refactor it for ESPEasy environment. These constants are device specific and should not be exported to clients. Can put them in a dedicated P164_data_struct_local.h.  But for me device specific data and code can be combined. Keeping them local also prevents clashes with other ESPEasy code.